src/puppetlabs/comidi.clj

```diff
@@ -167,11 +167,15 @@
     (update-in route-info [:path] conj "!")
 
     (sequential? pattern)
-    (if-let [next (first pattern)]
-      (update-route-info*
-        (update-in route-info [:path] conj next)
-        (rest pattern))
-      route-info)
+    (loop [info route-info
```

Comment:
I verified that once we have a sequential pattern, there is no reason to check these other conditions again, because when we recurse we always passed a new sequence. So this will skip some unnecessary stepping through the `cond`.